docs/new-feature-tutorial.rst

```diff
@@ -2,84 +2,178 @@
 New Feature Tutorial
 ====================
 
-.. attention::
-   This tutorial is an unfinished work -- contributions welcome!
+The changes needed to add a new feature will vary, of course, but this document
+provides a general outline of what you may need to do, as well as an example of
+the specific steps needed to add a new feature;
 
-The changes needed to add a new feature will vary, of course.  We give an
-example here that illustrates some of the common steps needed.  We describe
-the process of adding a new setting for admins that restricts inviting new
-users to admins only.
-
-Backend Changes
+General Process
 ===============
 
 Adding a field to the database
 ------------------------------
 
-The server accesses the underlying database in `zerver/models.py`.  Add
-a new field in the appropriate class, `realm_invite_by_admins_only`
-in the `Realm` class in this case.
+**Update the model:** The server accesses the underlying database in `zerver/
+models.py`. Add a new field in the appropriate class.
 
-Once you do so, you need to create the migration and run it; the
-process is documented at:
+**Create and run the migration:** The migration process is documented at:
```

Comment:
Probably we should put our own documentation, since this is just 2 ./manage.py commands in 99% of cases:

./manage.py makemigrations
./manage.py migrate